keanu-project/src/main/java/io/improbable/keanu/network/LambdaSection.java

```diff
@@ -90,18 +90,18 @@ public static LambdaSection getDownstreamLambdaSection(Vertex<?> aVertex, boolea
      * @param shouldAdd    true when a give vertex should be included in the result false otherwise
      * @return A Set of vertices that are in the direction implied by nextVertices and filtered by shouldAdd
      */
-    private static Set<Vertex> getVerticesDepthFirst(Vertex vertex,
-                                                     Function<Vertex, Collection<Vertex>> nextVertices,
-                                                     Predicate<Vertex> shouldAdd) {
+    public static Set<Vertex> getVerticesDepthFirst(Vertex vertex,
+                                                    Function<Vertex, Collection<Vertex>> nextVertices,
+                                                    Predicate<Vertex> shouldAdd) {
 
-        Set<Vertex> visited = new HashSet<>();
-        Deque<Vertex> stack = new ArrayDeque<>(nextVertices.apply(vertex));
+        Collection<Vertex> initialNext = nextVertices.apply(vertex);
+        Set<Vertex> queued = new HashSet<>(initialNext);
+        Deque<Vertex> stack = new ArrayDeque<>(initialNext);
         Set<Vertex> result = new HashSet<>();
         result.add(vertex);
 
         while (!stack.isEmpty()) {
             Vertex<?> visiting = stack.pop();
```

Comment:
Given we're using the Raw Type everywhere else here should we drop the type specifier here?

Reply:
👍